### PR TITLE
dashboard: fix not redirecting to login

### PR DESCRIPTION
**Pull request by koonpeng**
Preliminary fix for the dashboard not redirecting to login page. I tested using a local keycloak instance, will add instructions to run with keycloak later.

### CODE REVIEW

**Review by aaronchongth — CHANGES_REQUESTED** [review]
Just spotted an issue (tested on private windows as well) where after an initial login, if I close the window without logging out, and attempt to navigate to `baseURL/dashboard` again, nothing renders, so there is probably still something odd about the flow of routes and auth

if the user logs out manually, the URL works as expected

**Comment by koonpeng** [comment]
I'm not able to reproduce it, maybe it is due to some routing issues with trailing slashes? Have you tried going to `$baseUrl/dashboard/`?

File: packages/dashboard/examples/keycloak/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+This is an example using a local keycloak authentication provider.
+
+Follow the instrunctions below to run the example
```
[the remark this answers is not in the document]
Comment (koonpeng) [reply packages/dashboard/examples/keycloak/README.md]: thanks for the fixes! https://github.com/open-rmf/rmf-web/pull/1017/commits/182a0d98359eaeb63c8d4a1e4e4be5c2e7c19579

File: packages/dashboard/examples/keycloak/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+1. create a "dashboard" client
+1. create a rmf api server client scope
+1. create mappers for rmf api server client scopes
+1. link the dashboard client with the client scope
+1. export the public key
+
+3. Start rmf api server with the keycloak config
+
+```bash
+# be sure to run from packages/dashboard directory
```
[the remark this answers is not in the document]
Comment (koonpeng) [reply packages/dashboard/examples/keycloak/README.md]: https://github.com/open-rmf/rmf-web/pull/1017/commits/182a0d98359eaeb63c8d4a1e4e4be5c2e7c19579

File: packages/dashboard/examples/keycloak/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+1. Start a keycloak instance
+
+```bash
+docker run --rm -p 8080:8080 -e KEYCLOAK_ADMIN=admin -e KEYCLOAK_ADMIN_PASSWORD=admin quay.io/keycloak/keycloak:25.0.2 start-dev
+```
+
+2. Setup keycloak for rmf-web
+
+```bash
+examples/keycloak/keycloak-setup.bash -u admin -o rmf-web.pub
```
[the remark this answers is not in the document]
Comment (koonpeng) [reply packages/dashboard/examples/keycloak/README.md]: yeah that is intended, we don't want to run sudo commands that might break people's systems automatically

File: packages/dashboard/examples/keycloak/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+1. create a rmf api server client scope
+1. create mappers for rmf api server client scopes
+1. link the dashboard client with the client scope
+1. export the public key
+
+3. Start rmf api server with the keycloak config
+
+```bash
+# be sure to run from packages/dashboard directory
+RMF_API_SERVER_CONFIG="$(pwd)/examples/keycloak/api_server_config.py" pnpm -C ../api-server start
```
[the remark this answers is not in the document]
Comment (koonpeng) [reply packages/dashboard/examples/keycloak/README.md:33]: hmm I have the same version but I can't reproduce this.

File: packages/dashboard/examples/keycloak/README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (koonpeng) [reply packages/dashboard/examples/keycloak/README.md]: moved it to keycloak-example on dashboard package root. https://github.com/open-rmf/rmf-web/pull/1017/commits/182a0d98359eaeb63c8d4a1e4e4be5c2e7c19579

File: packages/dashboard/examples/keycloak/README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (aaronchongth) [reply packages/dashboard/examples/keycloak/README.md:33]: sorry for the delay, I've just tested it on another machine and it works just fine. Unsure yet what the root cause is, but I'll keep this in mind and point people to this thread if they encounter it. thanks!

File: packages/dashboard/src/components/rmf-dashboard.tsx [hunk 5]
```diff
@@ -153,12 +154,25 @@ export function RmfDashboard(props: RmfDashboardProps) {
   );
 
   const [userProfile, setUserProfile] = React.useState<UserProfile | null>(null);
+  const [authReady, setAuthReady] = React.useState(false);
```
Comment (aaronchongth) [inline packages/dashboard/src/components/rmf-dashboard.tsx:157]: This remains false after a page refresh, and doesn't get updated so nothing gets rendered. Could you help look into it?

**Review by aaronchongth — CHANGES_REQUESTED** [review]

File: packages/dashboard/src/components/rmf-dashboard.tsx (hunk 5 above)
Comment (koonpeng) [reply packages/dashboard/src/components/rmf-dashboard.tsx:157]: All the branches from the below effect ends with setting it to `true`, the only way I see for it to be false is that it is not finishing the promises, which indicate some timeout issues with kc or api server.

**Review by aaronchongth — APPROVED** [review]
LGTM!

File: packages/dashboard/src/components/rmf-dashboard.tsx (hunk 5 above)
Comment (aaronchongth) [reply packages/dashboard/src/components/rmf-dashboard.tsx:157]: thanks for confirming, the issue was a KC timeout

**Pull request merged**